### PR TITLE
aggregate assignment done

### PR DESCRIPTION
**Pull request by Gozied**
@basifat  pls review my assignment for aggregate functions

### CODE REVIEW

File: Abel/pf-project/project_assignment.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    total_gpa = sum(record['GPA'] for record in student_records.values()) 
+    return round(total_gpa, 2)
+
+def aggregate_2(student_records):
+    """ returns the sum of all student gpa in student records"""
+    total = 0
+    for record in student_records.values():
+        if record["GPA"] > 3.0:
+            total +=record["GPA"]
+    return round(total, 2)
```
Comment (basifat) [inline Abel/pf-project/project_assignment.py:26]: Lets create a variable gpa = record["GPC"]

File: Abel/pf-project/project_assignment.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+records = {
+              "1000": {'student_no': 1000, 'GPA': 4.9, 'full_name': 'James'}, 
+              "1001": {'student_no': 1001, 'GPA': 4.95, 'full_name': 'Tale'},
+              "1002": {'student_no': 1001, 'GPA': 2.9, 'full_name': 'Peter'}
+          }
+
+def aggregate(student_records):
+    total_gpa = sum(record['GPA'] for record in student_records.values())
```
Comment (basifat) [inline Abel/pf-project/project_assignment.py:17]: Update to as part of one-liner iterator.

File: Abel/pf-project/project_assignment.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    total = 0
+    for record in student_records.values():
+        if record["GPA"] > 3.0:
+            total +=record["GPA"]
+    return round(total, 2)
+
+def average_gpa(student_records):
+    """ calculete the aerage of all student gpa """
+    gpa_total= aggregate_2(student_records)
+    return gpa_total/len(student_records.values())
```
Comment (basifat) [inline Abel/pf-project/project_assignment.py:31]: Update len to include only necessary records.

File: Abel/pf-project/project_assignment.py (hunk 1 above)
Comment (Gozied) [reply Abel/pf-project/project_assignment.py]: assigned record["GPA"] to a variable

File: Abel/pf-project/project_assignment.py (hunk 2 above)
Comment (Gozied) [reply Abel/pf-project/project_assignment.py]: changes made to aggregate function

File: Abel/pf-project/project_assignment.py (hunk 3 above)
Comment (Gozied) [reply Abel/pf-project/project_assignment.py]: Updated  average_gpa function to return only necessary records

**Review by basifat — APPROVED** [review]

**Pull request merged**